cdap-operational-stats-core/src/main/java/co/cask/cdap/operations/hdfs/AbstractHDFSStats.java

```diff
@@ -39,4 +63,85 @@
   public String getServiceName() {
     return SERVICE_NAME;
   }
+
+  public String getWebURL() {
+    try {
+      if (HAUtil.isHAEnabled(conf, getNameService())) {
+        URL haWebURL = getHAWebURL();
+        if (haWebURL != null) {
+          return haWebURL.toString();
+        }
+      } else {
+        try (FileSystem fs = FileSystem.get(conf)) {
+          URL webUrl = rpcToHttpAddress(fs.getUri());
+          if (webUrl != null) {
+            return webUrl.toString();
+          }
+        }
+      }
+      lastCollectFailed = false;
+    } catch (Exception e) {
+      // TODO: remove once CDAP-7887 is fixed
+      if (!lastCollectFailed) {
+        LOG.warn("Error in determining HDFS URL. Web URL of HDFS will not be available in HDFS operational stats.", e);
+      }
+      lastCollectFailed = true;
+    }
+    return null;
+  }
+
+  @Nullable
+  private String getNameService() {
+    Collection<String> nameservices = conf.getTrimmedStringCollection(DFSConfigKeys.DFS_NAMESERVICES);
+    if (nameservices.isEmpty()) {
+      // we want to return null from this method if nameservices are not configured, so it can be used in methods like
+      // HAUtil.isHAEnabled()
+      return null;
+    }
+    if (1 == nameservices.size()) {
+      return Iterables.getOnlyElement(nameservices);
+    }
+    throw new IllegalStateException("Found multiple nameservices configured in HDFS. CDAP currently does not support " +
```

Comment:
This error message is misleading. It is only the operation stats collection doesn't support HDFS federation instead of CDAP itself.